src/vizier/services/cloud_connector/bridge/vzinfo.go

```diff
@@ -512,12 +512,9 @@ func (v *K8sVizierInfo) ParseJobYAML(yamlStr string, imageTag map[string]string,
 			imgPath := imgTag[0]
 
 			tagVers := semver.MustParse(val)
-
-			repoPath := publicImageRepo
-			if len(tagVers.Pre) > 0 {
-				repoPath = privateImageRepo
+			if len(tagVers.Pre) > 0 && strings.HasPrefix(imgPath, publicImageRepo) {
+				imgPath = strings.Replace(imgPath, publicImageRepo, privateImageRepo, 1)
```

Comment:
I don't think RCs even get pushed to the private repo anymore. So maybe we can just do away with this?